learning/prompts/documentation-lookup/31_advanced_batched_dofn.md

```diff
@@ -0,0 +1,35 @@
+Prompt:
+What is a Batched DoFn in Apache Beam?
+Response:
+In Apache Beam, a [Batched DoFn](https://beam.apache.org/documentation/programming-guide/#batched-dofns) is a [DoFn](https://beam.apache.org/documentation/programming-guide/#pardo) that can operate on batches of logical elements. These `DoFns` leverage popular Python vectorized libraries like [NumPy](https://numpy.org/), [SciPy](https://scipy.org/) and [pandas](https://pandas.pydata.org/) to efficiently process batches of elements in a `PCollection` rather than processing each element individually.
+
+Batched `DoFns` are only available in the Apache Beam Python SDK.
+
+You can provide element-wise, batched or both implementation for a `DoFn`
```

Comment:
```suggestion
To implement a batched ‘DoFn’, you define a [‘process_batch’](https://beam.apache.org/releases/pydoc/current/_modules/apache_beam/transforms/core.html#DoFn.process_batch) method instead of the typical [‘process’](https://beam.apache.org/releases/pydoc/current/_modules/apache_beam/transforms/core.html#DoFn.process) method used in regular ‘DoFn’s. The ‘process_batch’ method takes a batch of elements as input and produces a batch of elements as an output. It is important to note that batches must have a single set of timing properties (event time, window, etc.) for every logical element within the batch; batches cannot span multiple timestamps. 
```

Reply:
Suggested edits to explain what is a batched DoFn first, before jumping to "mixed" DoFns